doc/manual/src/command-ref/nix-shell.md

```diff
@@ -297,3 +297,9 @@ with import <nixpkgs> {};
 
 runCommand "dummy" { buildInputs = [ python pythonPackages.prettytable ]; } ""
 ```
+
+Aside from the very first line, which is a directive to the operating system,
+the `#! nix-shell` lines do not need to be at the beginning of the file. This
+can be useful for working around syntactic restrictions in languages like
+ECMAScript, Erlang, PHP, and Ruby, where the `#! nix-shell` lines must be
+hidden inside comments.
```

Comment:
This lacks motivation. I didn't understand why this information is worth knowing without reading the linked issue.

```suggestion
The script's file name is passed as the first argument to the interpreter specified by the `-i` flag.

Aside from the very first line, which is a directive to the operating system, the additional `#! nix-shell` lines do not need to be at the beginning of the file.
This allows wrapping them in block comments for languages where `#` does not start a comment, such as ECMAScript, Erlang, PHP, or Ruby.
```

Please [wrap lines at sentence boundaries](https://hydra.nixos.org/build/267989860/download/1/manual/development/documentation.html#markdown-conventions), this makes review suggestions less noisy for you and less cumbersome for us.